app/main.py

```diff
@@ -1,6 +1,15 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    archive_functions = {}
```

Comment:
You could store only args as a key for the inner cache. Rename your variable name and reformat the code to encounter that.